### PR TITLE
Feature/migrate to nx latest

### PR DESCRIPTION
**Pull request by fmmenchi**
This work:

- migrate to nx v11
- remove completely @angular-devkit
- add skipFormat option to generator-api-lib
- add skipFormat option to generator-api-spec

### CODE REVIEW

File: packages/nx-plugin-openapi/src/executors/generate-api-lib-sources/schema.d.ts [hunk 1]
```diff
@@ -1,6 +1,6 @@
-import { JsonObject } from '@angular-devkit/core';
+export interface BuildExecutorSchema {} // eslint-disable-line
```
Comment (trumbitta) [inline packages/nx-plugin-openapi/src/executors/generate-api-lib-sources/schema.d.ts:1]: Why do we need this empty interface?
If there's no direct substitute for the old `JsonObject` I'd say let's not extend anything, or let's copy the old `JsonObject` somewhere and keep extending it. 😄

**Review by trumbitta — CHANGES_REQUESTED** [review]
Something weird happening:

Using the api-spec generator, if you don't ask for a sample no lib directory will be created, only entries in `workspace.json` and `nx.json`.

image

File: packages/nx-plugin-openapi/src/executors/generate-api-lib-sources/schema.d.ts (hunk 1 above)
Comment (fmmenchi) [reply packages/nx-plugin-openapi/src/executors/generate-api-lib-sources/schema.d.ts:1]: I have removed the useless `BuildExecutorSchema` interface

**Comment by fmmenchi** [comment]
Do not merge please, i have noticed that  **(fix(generators-api-spec): create always sourceRoot directory** is wrong ... always create src dir even in dry-run mode

**Comment by trumbitta** [comment]
> I have added a `.gitkeep` file to force creation of `src` folder.
> 
> as far as i know Virtual file system tree (`@nrwl/tao/Tree`) APIs does not allow to add a dir but only to write files.
> 
> Do you know a way to achieve that without create the `.gitkeep`?

I think `.gitkeep` is a good solution: doesn't do any harm, reinforces the importance of the dir even when useless 👍 

Maybe the real Nx way was to always put the OpenAPI spec file inside `<mylib>/assets` to begin with, so that the still needed `.gitkeep` placeholder would go into a less "strange" dir.  
But asset dirs are a bit difficult to handle in libs 🤷 so I'm pretty content with your solution.

**Review by trumbitta — APPROVED** [review]

**Comment by trumbitta** [comment]
Hey @all-contributors let's add @fmmenchi his code contributions 🐗

**Pull request merged**